src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -45,25 +90,16 @@ constexpr char const* SPLIT_AND_RETRY_OOM_CLASS = "com/nvidia/spark/rapids/jni/S
 enum thread_state {
```

Comment:
Please use `enum class`. In C++, pure `enum` can be implicitly converted into `int` and vice versa.

Reply:
Of course we still can be **explicitly** converted from `enum class` to `int`, but that is always "explicit", i.e., intentional.